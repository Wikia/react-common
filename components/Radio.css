.wds-radio {
  display: inline-flex;
}
.wds-radio__input {
  display: none;
}
.wds-radio__label {
  align-items: center;
  cursor: pointer;
  display: inline-flex;
}
.wds-radio__label-text {
  line-height: normal;
}
.wds-radio__icon, .wds-radio__lock-icon {
  margin-right: 13px;
}
.wds-radio__input[disabled] + .wds-radio__label {
  color: #7f9998;
}