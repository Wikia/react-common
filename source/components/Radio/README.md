Each radio must have a value assigned to it.

Defaults:
```jsx
initialState = {
    checked: undefined
};

<div>
    <Radio
        value="one"
        checked={state.checked === "one"}
        onChange={() => setState({ checked: 'one' })}
     >
        One
    </Radio>
    <Radio
        value="two"
        checked={state.checked === 'two'}
        onChange={() => setState({ checked: 'two' })}
     >
        Two
    </Radio>
</div>
```

Disabled:
```jsx
<div>
    <Radio
        value="one"
        disabled
        checked={false}
        onChange={() => {}}
     >
        Disabled
    </Radio>
</div>
```

