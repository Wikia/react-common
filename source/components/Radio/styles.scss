@import "../../../node_modules/@fandom/design-system/dist/scss/wds-variables/index";

.wds-radio {
	display: inline-flex;

	&__input {
		display: none;
	}

	&__label {
		align-items: center;
		cursor: pointer;
		display: inline-flex;
	}

	&__label-text {
		line-height: normal;
	}

	&__icon,
	&__lock-icon {
		margin-right: 13px;
	}

	&__input[disabled] + .wds-radio__label {
		color: $wds-fandom-color-middle-gray;
	}
}
