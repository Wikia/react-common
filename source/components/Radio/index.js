import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import IconRadioActiveSmall from '../../icons/IconRadioActiveSmall';
import IconRadioEmptySmall from '../../icons/IconRadioEmptySmall';
import IconLockSmall from '../../icons/IconLockSmall';

import './styles.scss';

export default function Radio({ children, className, checked, disabled, ...restProps }) {
    return (
        // eslint-disable-next-line jsx-a11y/label-has-associated-control
        <label className={classNames('wds-radio', className)}>
            <input
                {...restProps}
                className="wds-radio__input"
                type="radio"
                checked={checked}
                disabled={disabled}
            />
            <span className="wds-radio__label">
                {checked && !disabled && <IconRadioActiveSmall className="wds-icon wds-icon-small wds-checkbox__icon" />}
                {!checked && !disabled && <IconRadioEmptySmall className="wds-icon wds-icon-small wds-checkbox__icon" />}
                {disabled && <IconLockSmall className="wds-icon wds-icon-small wds-checkbox__lock-icon" />}
                {children}
            </span>
        </label>
    );
}

Radio.propTypes = {
    /** Is radio button checked */
    checked: PropTypes.bool,
    /** Label passed as a child */
    children: PropTypes.node,
    /** Additional class name */
    className: PropTypes.string,
    /** Is radio button disabled */
    disabled: PropTypes.bool,
    /** Function triggered when checkbox changes its state */
    onChange: PropTypes.func.isRequired,
    /** Value that will be associated with radio */
    value: PropTypes.string.isRequired,
};

Radio.defaultProps = {
    checked: false,
    children: null,
    className: undefined,
    disabled: false,
};
