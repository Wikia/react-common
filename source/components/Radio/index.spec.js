import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import Radio from './index';

/* eslint-disable no-alert */

test('Radio renders correctly with default values', () => {
    const component = renderer.create(
        <Radio value="val" onChange={() => {}} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
});

test('Radio renders correctly checked (and with label and class name)', () => {
    const component = renderer.create(
        <Radio value="val" checked onChange={() => {}} className="checked-box">a Radio</Radio>,
    );
    expect(component.toJSON()).toMatchSnapshot();
});

test('Radio renders correctly when disabled', () => {
    const component = renderer.create(
        <Radio value="val" disabled onChange={() => {}}>disabled radio</Radio>,
    );
    expect(component.toJSON()).toMatchSnapshot();
});


test('Radio is not clickable when disabled', () => {
    const mockonChange = sinon.spy();
    const wrapper = shallow(
        <Radio value="val" disabled onChange={mockonChange}>disabled</Radio>
    );

    wrapper.find('input').simulate('click');

    expect(mockonChange.notCalled).toBe(true);
});

test('Radio triggers onChange event', () => {
    const mockonChange = sinon.spy();
    const wrapper = shallow(
        <Radio value="val" onChange={mockonChange}>clickable</Radio>
    );

    wrapper.find('input').simulate('change');

    expect(mockonChange.calledOnce).toBe(true);
});
