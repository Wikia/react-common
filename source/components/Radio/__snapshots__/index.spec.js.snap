// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders correctly checked (and with label and class name) 1`] = `
<label
  className="wds-radio checked-box"
>
  <input
    checked={true}
    className="wds-radio__input"
    disabled={false}
    onChange={[Function]}
    type="radio"
    value="val"
  />
  <span
    className="wds-radio__label"
  >
    <div
      className="wds-icon wds-icon-small wds-radio__icon"
    />
    <span
      className="wds-radio__label-text"
    >
      a Radio
    </span>
  </span>
</label>
`;

exports[`Radio renders correctly when disabled 1`] = `
<label
  className="wds-radio"
>
  <input
    checked={false}
    className="wds-radio__input"
    disabled={true}
    onChange={[Function]}
    type="radio"
    value="val"
  />
  <span
    className="wds-radio__label"
  >
    <div
      className="wds-icon wds-icon-small wds-radio__lock-icon"
    />
    <span
      className="wds-radio__label-text"
    >
      disabled radio
    </span>
  </span>
</label>
`;

exports[`Radio renders correctly with default values 1`] = `
<label
  className="wds-radio"
>
  <input
    checked={false}
    className="wds-radio__input"
    disabled={false}
    onChange={[Function]}
    type="radio"
    value="val"
  />
  <span
    className="wds-radio__label"
  >
    <div
      className="wds-icon wds-icon-small wds-radio__icon"
    />
    <span
      className="wds-radio__label-text"
    />
  </span>
</label>
`;
